### ISSUE TITLE
Incoming V2 - How to talk about it and beta testers

### ISSUE THREAD

**Issue by PapyElGringo** [PapyElGringo/material-awesome#40]
Hello everyone !

Im currently working hard on the next step of the project and very soon it's will be ready to be beta tested before the big release. 

But unfortunately I don't know how to communicate with you guys, I'll start with an issue here but we need to found a better way for me to push some informations. **Any idea??** 

### Why a V2 and What is inside?
When I released Material Awesome I had 3 frustrations: 
The first was the lack of animations. The Material design guidelines are full of delightful animations and I wanted to have some to help understand how the system work and to propose a nicer interface. 
The second was the difficulty to install. I had a tons of messages telling me that the project was awesome and they would like to have a system like that but they had no clue how to install it or AwesomeWM. And after working hard on this I wanted it to be available for the maximum of peoples.
The last frustration was the feeling that I had to redevelop all the basic features of the system If i wanted them to be well integrated with my system. Like a calendar or a wifi picker like basic stuff. I would have prefer to focus on the desktop UX that's it.

So I started the V2: **Material Shell**.

A Gnome-shell extension bringing all the features of Awesome Material but directly inside GNOME and available in 2 clicks ! GNOME is the most used system in the linux community and provide a nice Javascript Animation API. And a full featured desktop with a high level of customization. It's was perfect for the evolution of the project.

But it's required basically a complete rewrite of the project and much more. I never planned to work so much time on this project but I had a tons of good feedback on this and I want to bring the best experience possible !

The Beta should be available later this month and I will need some help to test it and fix the most obvious bugs before releasing it.

Thanks to all of you guys. If you a reading this lines, know that I worked hard for you and it's was a pleasure !

**Comment by omark96** [comment]
Maybe a discord server? It's a good way to be able to make announcements and also for people to post questions and such.

Oh and I'm really interested in this v2, definitely gonna give it a go.

**Comment by aislanmaia** [comment]
I really liked the layout of MW v1. Do you think is possible to maintain that? And I don't know the support of Gnome for window tiles (and keyboard shortcuts), that was the basic and more appreciated feature of AwesomeWM + Material. :disappointed:

**Comment by PapyElGringo** [comment]
@aislanmaia You don't have to be worry about that. I am bringing all the most important feature we got to GNOME that's mean the layout, the workflow and of course the windows tiling.
It's should only be better in every way. It's just asked a lot of worked and I've done it !

**Comment by aislanmaia** [comment]
@PapyElGringo let us know how we can help you with the dev, since its now related to Gnome (GTK) things..  maybe, we need to learn GTK in the process to help to development this?

**Comment by PapyElGringo** [comment]
Material-shell is a gnome-shell extension and it's all developed in Javascript Wis GJS and Clutter and we don't need to know about gtk. The main difficulty for me was to understand how the gnome-shell worked since the documentations about it are nearly inexistent so I studied his code source for a long time before being able to do what I wanted!

**Comment by PapyElGringo** [comment]
A discord server seem yo be a good idea since Slack is mainly for private group! Any other ideas?

**Comment by nununoisy** [comment]
One suggestion I have is more distro support. When I installed v1 I had to change a lot since Debian is not at all like Manjaro. One example was the package updater - I had to modify commands and Lua patterns to conform to `apt-get` instead of `pacman`. Someone who doesn’t know how everything in their system works will be confused when things don’t just work. Each module should have interchangeable functions that can be changed to adapt to a different distro, and some type of extensibility to allow users to add extra.

Another is a settings app, like what is in GnomeDE. Diving into a Lua file to change what app opens when you press the + button is cumbersome at best and difficult at worst. It should change things like that, but also handle the operations done by `wicd`, `blueberry`, etc.

Those two things would make Material Awesome way more awesome.

**Comment by aislanmaia** [comment]
Hey @nununoisy since the project is moving to be a Gnome extension, I think the things in Lua doesn't apply anymore.

**Comment by PapyElGringo** [comment]
@nununoisy Indeed the v1 was made specifically for my config so you had to adapt it to your system but in the v2 it's will all be manage by GNOME it's self ! We only need to adapt the UI and the UX to match our needs.
Anyway the V2 has been think to be installed in 2 clicks !

**Comment by nununoisy** [comment]
I get it now! Sorry for the confusion.

In that case, would there be a way to integrate some settings into the GNOME suite (i.e. within the Settings app shipped with GNOME, or possibly under `gnome-tweak-tool`)?

The extensibility no longer matters though - GNOME handles all of it.

**Comment by PapyElGringo** [comment]
Yeah at the end we should have a settings page in the tweak tool to handle preferences.

I created the Discord if you want to join. But it's not configured at all for the moment :D

https://discord.gg/QS6ccCY

**Comment by zeroryuki** [comment]
the discord link expired ATM.. would like to hear the update about this even i recently move from using the DM especially gnome

**Comment by PapyElGringo** [comment]
Strange... The discord link should be still usable but here a new one https://discord.gg/vBb7D9a.
And for the update of the project. The Alpha should be available this week and I will need some alpha testers to try it !
